pkg/validations/cluster.go

```diff
@@ -243,3 +243,15 @@ func ValidateEksaReleaseExistOnManagement(ctx context.Context, k kubernetes.Clie
 	}
 	return nil
 }
+
+// ValidatePauseAnnotation checks if the target cluster has annotation anywhere.eks.amazonaws.com/paused set to true or not.
+func ValidatePauseAnnotation(ctx context.Context, k KubectlClient, cluster *types.Cluster, clusterName string) error {
+	currentCluster, err := k.GetEksaCluster(ctx, cluster, clusterName)
+	if err != nil {
+		return err
+	}
+	if currentCluster.IsReconcilePaused() {
+		return fmt.Errorf("remove cluster controller reconciliation pause annotation %s before upgrading the cluster %s", currentCluster.PausedAnnotation(), clusterName)
```

Comment:
This sounds like remediation message instead of an error message, is it better to change it to something like "paused controller cannot be upgraded...?". Others look good.